pulumi/backuprunner/src/cleanup.ts

```diff
@@ -0,0 +1,56 @@
+import k8s, { KubernetesObject } from '@kubernetes/client-node'
+
+export const cleanup = async (k8sApi: k8s.KubernetesObjectApi, sts: string, namespace: string, pvcList: string, backupCount: number) => {
+  var pvcCount = pvcList.split(',').length;
+  var pvcsToKeepCount = pvcCount*backupCount;
+
+  const snapshots = await getExistingVolumeSnapshots(k8sApi, sts, namespace);
+  console.log(`Backup count to keep - ${backupCount}. ${sts} consists of ${pvcCount} PVC's, so will keep ${pvcsToKeepCount} latest snapshots`)
+
+  if(snapshots.length <= pvcsToKeepCount){
+    console.log(`Not archiving old snapshots, too few results`)
+  }else {
+    await removeOldSnapshots(k8sApi, snapshots, pvcsToKeepCount)
+  }
+
+  console.log("Snapshot cleanup completed");
+}
+
+const getExistingVolumeSnapshots = async (k8sApi: k8s.KubernetesObjectApi, sts: string, namespace: string): Promise<KubernetesObject[]> => {
+  const snapshots = await k8sApi.list("snapshot.storage.k8s.io/v1", "VolumeSnapshot", namespace, undefined, undefined, undefined, undefined, `statefulset=${sts}`);
+  const items = snapshots.body.items
+  console.log(`Found ${items.length} snapshots for ${namespace}.${sts}`)
+  return items.sort(sortByCreationTimestampDesc)
+}
+
+const sortByCreationTimestampDesc = ((a: KubernetesObject, b: KubernetesObject) => (new Date(b.metadata!!.creationTimestamp!!).getTime()  - new Date(a.metadata!!.creationTimestamp!!).getTime()));
```

Comment:
typically don't like to see `!!` but in this case I am ok since it is a very defined use case

Reply:
Agreed, couldn't find a better way but with the change to the interface from down below it might be avoidable

Reply:
Yeah I tested it out locally and that should give us the explicit typing needed to avoid this. Thought about it after this comment haha